unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -607,6 +608,42 @@ public static IntPtr unity_method_get_function_pointer([NativeCallbackType("Mono
         return t.IsValueType;
     }
 
+    private static ConcurrentDictionary<Type, bool> s_isBlittableCache = new ConcurrentDictionary<Type, bool>();
+    [return: NativeCallbackType("gboolean")]
+    public static bool class_is_blittable(
+        [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        bool isBlittable = false;
+        Type t = klass.TypeFromHandleIntPtr();
+
+        if (s_isBlittableCache.TryGetValue(t, out isBlittable))
```

Comment:
If we made the `s_isBlittableCache` take the `IntPtr` as a key we could skip the `TypeFromHandleIntPtr` call.